lib/devices/Viomi.js

```diff
@@ -80,85 +162,4 @@ const ERROR_MAP = Object.freeze({
     2103: "CHARGING"
 });
 
-Viomi.prototype.updateStatus = function(newState) {
-    if ("run_state" in newState) {
-        const oldStateFastPull = MiioVacuum.FAST_MAP_PULLING_STATES.indexOf(this.status.state);
-        this.status.state = STATUS_MAP[newState["run_state"]];
-        this.status.human_state = this.status.state;
-        const newStateFastPull = MiioVacuum.FAST_MAP_PULLING_STATES.indexOf(this.status.state);
-        if (newStateFastPull && !oldStateFastPull)
-            this.pollMap();
-    }
-    if ("err_state" in newState) {
-        let code = newState["err_state"]; // TODO adopt to viomi codes
-        if (ERROR_MAP[code] !== undefined) {
-            code = ERROR_MAP[code];
-            this.status.error_code = code;
-            if (code != "CHARGING" && code != "CHARGE_COMPLETED")
-                this.status.human_error = code;
-        } else {
-            this.status.error_code = code;
-            this.status.human_error = "Error_" + this.status.error_code;
-        }
-    }
-    if ("battary_life" in newState)
-        this.status.battery = newState["battary_life"];
-    if ("s_time" in newState)
-        this.status.clean_time = newState["s_time"];
-    if ("s_area" in newState)
-        this.status.clean_area = newState["s_area"] * 1000 * 1000; // m² to mm²
-    this.events.emit("miio.status", this.status);
-};
-
-Viomi.prototype.pollStatus = function(callback) {
-    this.sendLocalWithPromise("get_prop", STATE_PROPERTIES).then((res) => {
-        let statusDict = {};
-        STATE_PROPERTIES.forEach((key, index) => statusDict[key] = res[index]);
-        this.updateStatus(statusDict);
-        callback();
-    });
-};
-
-Viomi.prototype.getCurrentStatus = function() {
-    return Promise.resolve(this.status);
-};
-
-Viomi.prototype.pollMap = function() {
-    clearTimeout(this.pollMapTask);
-    this.sendLocalWithPromise("set_uploadmap", [2], {timeout: 2000}).finally(() => {
```

Comment:
Hmm sendLocalWithPromise should have these options too, but then we also need to support this for the cloudSocket… I can do that another time though.